bootstrap.py

```diff
@@ -218,6 +218,45 @@ def install_prereqs():
         delete_file('/etc/yum.repos.d/katello-client-bootstrap-deps.repo')
 
 
+def get_puppet_path():
+    """
+    Get the path of where the puppet excutable is located.
+    """
+    puppet_lookup_paths = [
+        "/usr/bin",
+        "/opt/puppetlabs/puppet/bin"
+    ]
+    executable_path = None
+    for path in puppet_lookup_paths:
+        executable_path = os.path.join(path, 'puppet')
+        if os.path.isfile(executable_path):
+            break
+        executable_path = None
+
+    if executable_path is None:
+        print_error("No puppet executable found")
+        sys.exit(1)
+    return executable_path
+
+
+def get_puppet_version():
```

Comment:
I think I'd still prefer using the RPMdb like this:

```python
def get_puppet_version():
    ts = rpm.TransactionSet()
    for name in ['puppet', 'puppet-agent']:
        mi = ts.dbMatch('name', name)
        for h in mi:
            major = h['version'].split('.')[0]
            if h['name'] == 'puppet-agent' and int(major) == 1:
                puppet_version = 4
            else:
                puppet_version = major
            return puppet_version
```

This works fine for Puppet 3, 4 and 5

@sideangleside whatcha think?

Reply:
@evgeni If I am not getting wrong, we are comparing the puppet-agent packaging version here. I doubt if this will be future proof or not, in case packaging convention changes in future for the newer releases of Puppet.

Reply:
correct, it's using the packaging version. I would not expect the major version of the packaging to differ from the major version of the agent in the future (well, puppet4 agent is an exemption here :()

Reply:
I am strongly supportive of checking the RPMs version number as it is consistent with other checks for packages that we do. Additionally, checking via an executable's location is just as prone to packaging changes as anything else, so i'd want to be consistent.